### PR TITLE
ref(banner): Collocate banner specific logic

### PR DESCRIPTION
**Pull request by evanpurkhiser**
- Moves the localstorage dismissing functionality into the base banner
 - Moves the media query size check into the discover (eventsv2) banner

### CODE REVIEW

File: static/app/views/eventsV2/landing.tsx [hunk 1]
```diff
@@ -55,15 +48,11 @@ type Props = {
 } & AsyncComponent['props'];
 
 type State = {
-  isBannerHidden: boolean;
-  isSmallBanner: boolean;
```
Comment (EvanPurkhiser) [inline static/app/views/eventsV2/landing.tsx:59]: This state does not belong in this component

File: static/app/views/eventsV2/landing.tsx [hunk 2]
```diff
@@ -232,24 +189,12 @@ class DiscoverLanding extends AsyncComponent<Props, State> {
...
 
-    return (
-      <Banner
-        organization={organization}
-        resultsUrl={resultsUrl}
-        isSmallBanner={this.state.isSmallBanner}
-        onHideBanner={this.handleBannerClick}
-      />
-    );
+    return <Banner organization={organization} resultsUrl={resultsUrl} />;
```
Comment (EvanPurkhiser) [inline static/app/views/eventsV2/landing.tsx:197]: Banner has learned to do all of the above

File: static/app/views/eventsV2/savedQuery/index.tsx [hunk 3]
```diff
@@ -161,7 +161,7 @@ class SavedQueryButtonGroup extends React.PureComponent<Props, State> {
       (savedQuery: SavedQuery) => {
         const view = EventView.fromSavedQuery(savedQuery);
 
-        setBannerHidden(true);
+        Banner.dismiss('discover');
```
Comment (EvanPurkhiser) [inline static/app/views/eventsV2/savedQuery/index.tsx:164]: Banner is now responsible for dismissing, so I added this helper method onto the Banner prototype.

File: static/app/views/eventsV2/banner.tsx [hunk 4]
```diff
@@ -99,14 +100,16 @@ function DiscoverBanner({organization, resultsUrl, isSmallBanner, onHideBanner}:
     });
   }
 
+  const isSmallBanner = useMedia(`(max-width: ${theme.breakpoints[1]})`);
```
Comment (EvanPurkhiser) [inline static/app/views/eventsV2/banner.tsx:103]: Moved the theme stuff into here with this nice little `react-use`

**Comment by billyvg** [comment]
Are there tests for `<Banner>`?

**Comment by EvanPurkhiser** [comment]
I'll add some

**Review by markstory — COMMENTED** [review]
I like the cleanup around the Banner component, and usage of custom hooks. I'm not sure that we need to pull in all these dependencies to have the functionality though.

**Comment by EvanPurkhiser** [comment]
Remove react-use, inlined useMedia, and just wrote a simple useDismissed

**Review by scefali — APPROVED** [review]
LGTM

**Pull request merged**